agent/acl.go

```diff
@@ -187,23 +190,23 @@ func (a *Agent) vetCheckRegisterWithAuthorizer(authz acl.Authorizer, check *stru
 	// Vet the check itself.
 	if len(check.ServiceName) > 0 {
 		if authz.ServiceWrite(check.ServiceName, &authzContext) != acl.Allow {
-			return acl.ErrPermissionDenied
+			return acl.PermissionDeniedWithCause("Missing service:write for service %s", check.ServiceName)
 		}
 	} else {
 		if authz.NodeWrite(a.config.NodeName, &authzContext) != acl.Allow {
-			return acl.ErrPermissionDenied
+			return acl.PermissionDeniedWithCause("Missing write permission for node %s", a.config.NodeName)
```

Comment:
Could you change that to `Missing node:write for node %s` so that it is consistent with the service messages?

Reply:
Hi @i0rek 

Thanks for reviewing!

I've rebased and changed error messages. 

I'm not sure if I used correct values for serviceName. We used `structs.ServiceIDString(existing.ServiceName, &existing.EnterpriseMeta)`, to display id in error msg, but I used `CompoundServiceName` for getting  ServiceName.